File: TP01/TP01Q07 - Is em C/Is.c
//TP01Q07 - is em c
//@Author: Vitoria Simil Araujo

#include "stdio.h"
#include "string.h"
#include "stdbool.h"

//functions
bool isEnd(char input[]);
bool isLetter(char c);
bool isVowel(char c);
bool isDigit(char c);
char* isAllVowels(char input[]);
char* isAllConsonats(char input[]);
char* isInteger(char input[]);
char* isDouble(char input[]);
void execute(char input[]); 

int main(void) {
    char input[1000] = {};

    //read the input archive
    scanf("%[^\n]", input);
    getchar();
    while ( !isEnd(input) ) {
        execute(input);
        scanf("%[^\n]", input);
        getchar();
    }
}

//method to execute
void execute(char input[]) {
    printf("%s%s%s%s\n", isAllVowels(input), isAllConsonats(input), isInteger(input), isDouble(input));
}

//method to verify if a string equals FIM
bool isEnd(char input[]) {
    return (input[0] == 'F' && input[1] == 'I' && input[2] == 'M');
}

//method to verify if a char equals a letter
bool isLetter(char c) {
    //data
    int code = (int) c;

    return (( code >= 65 && code <= 90 ) || ( code >= 97 && code <= 121 ));
}

//method to verify if a char equals a vowel
bool isVowel(char c) {
    //data
    int vowels[10] = {65, 69, 73, 79, 85, 97, 101, 105, 111, 117};

    for (  int i = 0; i < 10; i++ ) {
        if ( vowels[i] == (int) c ) return true;
    }

    return false;
}

//method to verify if a char equals a digit
bool isDigit(char c) {
    //data
    int code = (int)c;

    return ( code >= 48 && code <= 57);
}

//method to verify if a string is composed only by vowels
char* isAllVowels(char input[]) {
    for( int i = 0; i < strlen(input) - 1; i++ ) {
        if (!(isVowel(input[i]))) return "NAO ";
    }

    return "SIM ";
}

//method to verify if a string is composed only by consonats
char* isAllConsonats(char input[]) {
    for ( int i = 0; i < strlen(input) - 1; i++ ) {
        if ( !(isLetter(input[i])) || (isVowel(input[i])) ) return "NAO ";
    }

    return "SIM ";
}

//method to verify if a string equals an integer number
char* isInteger(char input[]) {
    for ( int i = 0; i < strlen(input); i ++) {
        if ( !isDigit(input[i])) {
            return "NAO ";
        }
    }

    return "SIM ";
}

//method to verify if a string equals a double number
char* isDouble(char input[]) {
    //data
    int separators = 0;

    for ( int i = 0; i < strlen(input) - 1; i++ ) {
        if ( !isDigit((input[i]))) {
            if ( (input[i]) == ',' || (input[i]) == '.' ) separators++;
            else return "NAO";
        }
    }

    return ( separators == 0 || separators == 1 ) ? "SIM" : "NAO";
}

File: TP01/TP01Q15 - RECURSIVO - Is em Java/Is.java
//TP01Q15 - is em java iterativo
//@Author: Vitoria Simil Araujo

class Is {
    private static boolean isEnd(String input) {
        return (input.charAt(0) == 'F' && input.charAt(1) == 'I' && input.charAt(2) == 'M');
    }

    private static boolean isVowel(char c) {
        //data
        int vowels[] = {65, 69, 73, 79, 85, 97, 101, 105, 111, 117};

        for (  int i = 0; i < 10; i++ ) {
            if ( vowels[i] == (int) c ) return true;
        }

        return false;
    }

    //verify if a string is all composed by vowels
    private static String isVowel(int index, String input) {
        if ( index < input.length() ) {
            char c = input.charAt(index);

            if ( (isVowel(c)) ) {
                return isVowel(index + 1, input);
            }
            else {
                return "NAO ";
            }
        }
        else {
            return "SIM ";
        }
    }

    //verify if a String is all composed by consonants
    private static String isConsonant(int index, String input) {
        if ( index < input.length() ) {
            int c = (int)input.charAt(index);

            if ( (c >= 65 && c <= 90) && ( c >= 97 && c <= 121 ) && (isVowel(input.charAt(index)))) {
                return isConsonant(index + 1 , input);
            }
            else {
                return "NAO ";
            }
        }
        else {
            return "SIM ";
        }
    }

    //verify if a string is all composed by digits
    private static String isDigit(int index, String input) {
        if ( index < input.length() ) {
            int c = (int)input.charAt(index);

            if ( c >= 48 && c <= 57 ) {
                return isDigit(index + 1 , input);
            }
            else {
                return "NAO ";
            }
        }
        else {
            return "SIM ";
        }
    }

    //verify if a string is a real number  
    private static int isDouble(int index, String input) {  
        int separators = 0;

        if ( index < input.length() ) {
            int c = (int)input.charAt(index);
            char character = input.charAt(index);

            if ( c >= 48 && c <= 57 ) {
                return separators + isDouble(index + 1 , input);
            }
            else if ( character == ',' || character == '.') {
                separators++;
                return separators + isDouble(index + 1 , input);
            }
            else {
                return 2;
            }
        }
        else {
            return separators;
        }
    }

    //method to execute
    public static String execute(String input) {
        //data
        String answer = "";
        int separators = 0;
        
        answer += isVowel(0, input);
        answer += isConsonant(0, input);
        answer += isDigit(0, input);
        separators = isDouble(0, input);
        answer += (separators <= 1 ) ? "SIM" : "NAO";
        
        return answer;
    }

    public static void main(String[] args) {
        //data
        String input = "";

        //read the input archive
        input = MyIO.readLine();

        while ( !isEnd(input) ) {
            System.out.println(execute(input));
            input = MyIO.readLine();
        }
    }
}